content/en/platform/corda/5.0/key-concepts/cluster-admin/tech-stack/_index.md

```diff
@@ -69,7 +69,18 @@ Corda requires the following types of keys:
 * Ledger
 * CorDapp publisher code signing
 
-All of the above keys are stored in the Crypto databases (cluster and virtual nodes) and they are all encrypted at rest with “Wrapping keys”. <!--more info to follow-->
+All of these keys are stored in the Crypto databases (cluster and virtual nodes) and they are all encrypted at rest with “wrapping keys”. 
+
+{{< 
+  figure
+	 src="wrapping-keys.png"
+   width="75%"
+	 figcaption="Wrapping Keys"
+>}}
+
+The diagram illustrates that key wrapping is hierarchical. The master wrapping key protects other wrapping keys, such as the virtual node related keys, which in turn protect the private keys used by Corda. It must never be possible for someone with a copy of the Corda database, or a virtual node database, to decrypt the keys stored in the database using other information stored in the database. Threfore, the master wrapping key, or the information required to generate this key, must be stored and managed outside Corda. This can be acheived in one of the following ways:
```

Comment:
```suggestion
The diagram illustrates that key wrapping is hierarchical. The master wrapping key protects other wrapping keys, such as the virtual node related keys, which in turn protect the private keys used by Corda. It must never be possible for someone with a copy of the Corda database, or a virtual node database, to decrypt the keys stored in the database using other information stored in the database. Threfore, the master wrapping key, or the information required to generate this key, must be stored and managed outside Corda. This can be achieved in one of the following ways:
```